x-pack/plugins/cases/public/components/all_cases/columns_popover.test.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { TestProviders } from '../../common/mock';
+import { ColumnsPopover } from './columns_popover';
+
+describe('ColumnsPopover', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const selectedColumns = [
+    { field: 'title', name: 'Title', isChecked: true },
+    { field: 'category', name: 'Category', isChecked: false },
+  ];
+
+  it('renders correctly a list of selected columns', () => {
+    render(
+      <TestProviders>
+        <ColumnsPopover selectedColumns={selectedColumns} onSelectedColumnsChange={() => {}} />
+      </TestProviders>
+    );
+
+    userEvent.click(screen.getByTestId('column-selection-popover'));
+
+    selectedColumns.forEach(({ field, name, isChecked }) => {
+      expect(screen.getByTestId(`column-selection-switch-${field}`)).toHaveAttribute(
+        'aria-checked',
+        isChecked.toString()
+      );
+      expect(screen.getByText(name)).toBeInTheDocument();
+    });
+  });
+
+  it('clicking a switch calls onSelectedColumnsChange with the right params', async () => {
+    const onSelectedColumnsChange = jest.fn();
+
+    render(
+      <TestProviders>
+        <ColumnsPopover
+          selectedColumns={selectedColumns}
+          onSelectedColumnsChange={onSelectedColumnsChange}
+        />
+      </TestProviders>
+    );
+
+    userEvent.click(screen.getByTestId('column-selection-popover'));
+    userEvent.click(
+      screen.getByTestId(`column-selection-switch-${selectedColumns[0].field}`),
+      undefined,
+      { skipPointerEventsCheck: true }
+    );
+
+    await waitFor(() => {
+      expect(onSelectedColumnsChange).toHaveBeenCalledWith([
+        { ...selectedColumns[0], isChecked: false },
+        selectedColumns[1],
+      ]);
+    });
+  });
+
+  it.skip('drag and rop calls onSelectedColumnsChange with the right params', async () => {});
```

Comment:
Any ideas? I've only seen this being done with _enzyme_ to `simulate` the event. I tried using https://testing-library.com/docs/dom-testing-library/api-events/#createeventeventname but it did not work. I looked into the `eui` documentation but couldn't find any relevant drag-and-drop testing.